tests/store/filters/test_file_filters.py

```diff
@@ -82,3 +84,74 @@ def test_filter_files_by_id_returns_none_for_invalid_id(
     # THEN no file should be returned
     filtered_file: File = file_query.first()
     assert filtered_file is None
+
+
+def test_filter_files_by_tags_returns_correct_files(populated_store: Store):
+    """Test filtering files by tags."""
+
+    # GIVEN a store with files
+    file: File = populated_store._get_query(table=File).first()
+    tag_names: List[str] = [tag.name for tag in file.tags]
+
+    # WHEN filtering files by tags
+    filtered_files_query = filter_files_by_tags(
+        files=populated_store._get_join_file_tag_query(),
+        tag_names=tag_names,
+    )
+
+    filtered_files: List[File] = filtered_files_query.all()
+
+    # THEN each file should have all the requested tags
+    for filtered_file in filtered_files:
+        assert all(tag.name in tag_names for tag in filtered_file.tags)
+
+
+def test_filter_files_by_tags_returns_empty_list_when_no_files_match_tags(
+    populated_store: Store,
+):
+    """Test filtering files by tags when no files match the given tags."""
+
+    # GIVEN a store with files
+    # GIVEN a tag that does not exist in any files in the store
+    tag_name = "nonexistent_tag"
```

Comment:
```suggestion
    tag_name = "non_existent_tag"
```
Seems general enough to be fixture